### ISSUE TITLE
Evaluates the uncertainty of the surrogate model at points x

### ISSUE THREAD

**Issue by pvh1983** [dme65/pySOT#27]
Hello, 

Could you please give me an example that prints out "the uncertainty of the surrogate model at points x" as said in the online manual? 

predict_std(x): Evaluates the uncertainty of the surrogate model at points x

I am using your code to calibrate a highly non-linear groundwater model with a high number of model parameters (large than 40). The code worked perfectly and saved me a lot of time. I looking forward to see if I can use the code for uncertainty analysis related to estimated model parameters. I am using pysot with DYCORStrategy. 

Thanks, 
Hai.

**Comment by Ontahm** [comment]
-------------------------------------------------------------------------------------------------------------------
`gp = GPRegressor(dim=...)` # since you talk about uncertainty

> ` ... create strategy, run controller ... `

`gp.X` contains all points that are incorporated in the surrogate

`print(gp.predict_std(gp.X)[0,1]) `   # prints the uncertainty for the second point in gp.X


This prints the uncertainty for all points incorporated in the surrogate:
`[print("Uncertainty for x= "+ str(pt) +" is %.4f" %(gp.predict_std(pt)[0,0])) for pt in gp.X ]`

Hope this helps.

-------------------------------------------------------------------------------------------------------------------
Please close the issue.

**Comment by pvh1983** [comment]
Thanks @Ontahm. It worked perfect. Also, could you please explain me what is the meaning of predict_std? Is it the one standard deviation of the "best value found" (see attached figure)? 
Is there anyway that the code can print out the covariance matrix of estimated model parameters? Thanks. 


![image](https://user-images.githubusercontent.com/9260965/53717967-aa5a0000-3e0e-11e9-9a92-9356f1b3c6cb.png)

**Comment by dme65** [comment]
GPRegressor is just a simple wrapper class for the [GaussianProcessRegressor](https://scikit-learn.org/stable/modules/generated/sklearn.gaussian_process.GaussianProcessRegressor.html#sklearn.gaussian_process.GaussianProcessRegressor) in Sklearn. It may be more useful for your purpose to just use Sklearn directly. Their prediction method has the option of returning the full covariance matrix. I'm closing this for now as all of your questions should be easily answered by the Sklearn documentation.
